File: src/Models/Reservation.php
<?php

require_once __DIR__ . '/../Database.php';

class Reservation
{
    public static function create($data)
    {
        $db = Database::getConnection();
        $stmt = $db->prepare("
            INSERT INTO reservations (restaurant_id, user_id, reservation_date, reservation_time, code)
            VALUES (?, ?, ?, ?, ?)
        ");
        return $stmt->execute([
            $data['restaurant_id'],
            $data['user_id'],
            $data['reservation_date'],
            $data['reservation_time'],
            $data['code']
        ]);
    }

    public static function allByUser($userId)
    {
        $db = Database::getConnection();
        $stmt = $db->prepare("
            SELECT r.*, res.name AS restaurant_name
            FROM reservations r
            JOIN restaurants res ON res.id = r.restaurant_id
            WHERE r.user_id = ?
            ORDER BY r.reservation_date ASC
        ");
        $stmt->execute([$userId]);
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public static function deleteExpired()
    {
        $db = Database::getConnection();
        $stmt = $db->prepare("DELETE FROM reservations WHERE reservation_date < CURDATE()");
        $stmt->execute();
    }

    public static function delete($id, $userId)
    {
        $db = Database::getConnection();
        $stmt = $db->prepare("DELETE FROM reservations WHERE id = ? AND user_id = ?");
        return $stmt->execute([$id, $userId]);
    }

}

File: mobile_flutter/lib/models/reservation.dart
class Reservation {
  const Reservation({
    required this.id,
    required this.restaurantId,
    required this.userId,
    required this.reservationDate,
    required this.reservationTime,
    required this.code,
    this.restaurantName,
    this.username,
    this.email,
  });

  final int id;
  final int restaurantId;
  final int userId;
  final String reservationDate;
  final String reservationTime;
  final String code;
  final String? restaurantName;
  final String? username;
  final String? email;

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: (json['id'] as num?)?.toInt() ?? 0,
      restaurantId: (json['restaurant_id'] as num?)?.toInt() ?? 0,
      userId: (json['user_id'] as num?)?.toInt() ?? 0,
      reservationDate: (json['reservation_date'] ?? '') as String,
      reservationTime: (json['reservation_time'] ?? '') as String,
      code: (json['code'] ?? '') as String,
      restaurantName: json['restaurant_name'] as String?,
      username: json['username'] as String?,
      email: json['email'] as String?,
    );
  }
}
